frontend/model/state.js

```diff
@@ -108,37 +127,59 @@ const mutations = {
     Vue.delete(state.chatRoomScrollPosition, chatRoomId)
   },
   setChatRoomReadUntil (state, { chatRoomId, messageHash, createdDate }) {
-    const prevMentions = state.chatRoomUnread[chatRoomId].mentions
-    const prevOthers = state.chatRoomUnread[chatRoomId].others
-
+    // eslint-disable-next-line no-lone-blocks
+    {
+      // TODO: need to remove this block after we release 0.2.*
+      if (!state.chatRoomUnread[chatRoomId].messages) {
+        Vue.set(
+          state.chatRoomUnread[chatRoomId],
+          'messages',
+          [].concat(
+            (state.chatRoomUnread[chatRoomId].mentions || []).map(m => Object.assign({ type: MESSAGE_TYPES.TEXT }, m)),
+            (state.chatRoomUnread[chatRoomId].others || []).map(o => Object.assign({ type: MESSAGE_TYPES.INTERACTIVE }, o))
+          )
+        )
+      }
+      if (state.chatRoomUnread[chatRoomId].mentions) {
+        Vue.set(state.chatRoomUnread, chatRoomId, {
+          readUntil: { messageHash, createdDate, deletedDate: null },
+          mentions: [],
+          messages: state.chatRoomUnread[chatRoomId].messages
+            .filter(m => new Date(m.createdDate).getTime() > new Date(createdDate).getTime())
+        })
+        return
+      }
+    }
     Vue.set(state.chatRoomUnread, chatRoomId, {
       readUntil: { messageHash, createdDate, deletedDate: null },
-      mentions: prevMentions.filter(m => new Date(m.createdDate).getTime() > new Date(createdDate).getTime()),
-      others: prevOthers.filter(m => new Date(m.createdDate).getTime() > new Date(createdDate).getTime())
+      messages: state.chatRoomUnread[chatRoomId].messages
+        .filter(m => new Date(m.createdDate).getTime() > new Date(createdDate).getTime())
```

Comment:
These lines are the problem, `state.chatRoomUnread[chatRoomId].messages` is `undefined` when this is called.

So maybe this line needs some `?.` and `|| []` type stuff?